sympy/matrices/reductions.py

```diff
@@ -245,6 +248,38 @@ def complexity(i):
     return len(pivots)
 
 
+def _check_expr(M):
+    """Checks whether all the elements of the Matrix are Expr before converting to DomainMatrix.
+
+    Parameters
+    ==========
+
+    M:  Matrix
+
+    Returns
+    =======
+
+    Boolean
+        True if all elements are of type Expr, else False
+
+    """
+    types = set(map(type, M))
+    if all(issubclass(t, Expr) for t in types):
+        return True
+    return False
+
+def _eval_domain(M):
+    types = set(map(type, M))
+    if all(t.is_Rational for t in types):
+        return QQ
+
+    primitive_element = {p for p in M.atoms(Pow) if p.is_number and p.exp.is_Rational and p.exp.q != 1}
+    if len(primitive_element) == 1:
+        return True
+
+    return None
```

Comment:
This is a strange sort of function that returns either a domain or True or None.

Probably what would make most sense is to add an option to `DomainMatrix.get_domain` so that you can pass something like `radicals_limit=1` and then `get_domain` can check that argument and decide whether or not to pass `extension=True` to `construct_domain` depending on the number of radicals.